fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java

```diff
@@ -141,6 +143,26 @@ public String getDefaultVersion(String url, Class<? extends Resource> resourceTy
      *     if the resource type is not a definitional resource type
      */
     public <T extends Resource> T getResource(String url, Class<T> resourceType) {
+        return getResource(url, resourceType, null);
+    }
+
+    /**
+     * Get the resource for the given canonical url and resource type
+     *
+     * @param url
+     *     the canonical url (with optional version postfix)
+     * @param resourceType
+     *     the resource type
+     * @param providerName
+     *     the canonical class name of the provider that is calling get resources
```

Comment:
I was trying to decide how to make it clear *why* someone would want to call this (and pass their class name) instead of the other getResource call.

Maybe change the method name to reflect this purpose?
Or maybe just add on to the method description (to distinguish it from the other `getResource`)?
Or maybe update the parameter name to something like `providerNameToExclude`?

Reply:
I'll pick the last one `providerNameToExclude`